skrf/network.py

```diff
@@ -172,28 +170,25 @@
 
 import numpy as npy
 from numpy.linalg import inv as npy_inv
-from numpy import fft, gradient, ndarray, reshape, shape, ones, std
+from numpy import gradient, ndarray, reshape, shape, ones, std
 from scipy import stats, signal  # for Network.add_noise_*, and Network.windowed
 from scipy.interpolate import interp1d  # for Network.interpolate()
-import unittest  # fotr unitest.skip
 
+from .base_network import BaseNetwork
 from . import mathFunctions as mf
 from .frequency import Frequency
-from .util import get_fid, get_extn, find_nearest_index, slice_domain
+from .util import get_fid, get_extn, find_nearest_index, plotting_available
 from .time import time_gate
 
 from .constants import NumberLike, ZERO, K_BOLTZMANN, T0
 from .constants import S_DEFINITIONS, S_DEF_DEFAULT
 
+from .plotting import PlottingMixin
+
 if TYPE_CHECKING:
     import pandas as pd
 
-#from matplotlib import cm
-#import matplotlib.pyplot as plt
-#import matplotlib.tri as tri
-#from scipy.interpolate import interp1d
-
-class Network:
+class Network(PlottingMixin, BaseNetwork):
```

Comment:
To understand your approach, what is the purpose of this new BaseNetwork ABC class? Properly separate the base properties from the plotting ones?

Reply:
I thought about creating a base class, where Network and NetworkSet could inherit from. But I'm not sure yet. As said, NetworkSet improvements are WIP.